### ISSUE TITLE
unexpected '/' prepended to search pattern passed via command line option '-p', sometimes

### ISSUE THREAD

**Issue by ryenus** [gwsw/less#101]
I have a little git helper utility to show git release notes, in which I use the following less command to match the leading '*' characters:

    less -NRwXj9 -p '(?<=^ )\*(?= )' +jgn
see https://github.com/ryenus/git-relnotes/blob/master/git-relnotes#L18

But sometimes it fails to do the search even there're actually matches, resulting with a "Pattern not found" error.

By inspecting the last searched pattern, it appears that the search was executed with a leading '/' unexpectedly prepended to the search pattern.

### How to reproduce
1. ~~run `git-relnotes` from the linked source~~
2. ~~now search for something other than the pattern specified via the '-p' option, such as 'abc'~~
3. ~~quit less,~~ and re-run `git-relnotes 2.29.2`, at this point, there's only one matching `*` in the file, and it has been _consumed_ by `+jgn`
4. press `n` once, and **only once**, see that the message **Pattern not found  (press RETURN)** appears.
5. press `/`
6. at this point you would see that `/` is NOT starting a new search **<~~ THE BUG**, and if search highlight is on, you would see the previous matching `*` is no longer highlighted.
7. press `/`, then the up key, now I see a leading `/' was mistakenly prepended to the search pattern.

BTW, I think the pattern used with '-p' can be anything if not the one from the linked script, just make sure in step 2, the other pattern is not the same. Basically you can run less with any input and/or search patterns.

~~My guess is that when there're both a recent search pattern executed with in less, and a search pattern passed in via the '-p' option, less makes the mistake and prepends the '/' to the search pattern specified via the command line options.~~

### Version Info
less 563 (PCRE regular expressions), installed via Homebrew on macOS

**Comment by gwsw** [comment]
I'm sorry, I'm having trouble reproducing this. Starting with an empty .lesshst file, when I run your script, it displays a 4 line file and says "No previous regular expression", then "Pattern not found". I search for "abc", quit and run the script again, and it says "Nothing to search" and "Pattern not found". At that point there are 3 items in the search history, the first and third being your pattern and the second is "abc". I don't see an extraneous slash appear. Can you provide a simpler reproduction method, maybe using the less source files rather than a downloaded file?

**Comment by ryenus** [comment]
@gwsw following your comment I figured something and updated the reproducing steps above.

So it appears when less says "**Pattern not found  (press RETURN)**", there're 2 cases:
1. there's no matches at all
2. there has been matches, but all consumed by previous `n`, or `N` actions.

Though in both cases, with the immediate subsequent `n`, the message **Pattern not found  (press RETURN)** would appear. but in the second case, pressing `/` would cause the bug, as explained in step 6 above.


Meanwhile, there's another issue, under the following conditions, 
   * ~/.lesshst doesn't exist
   * a pattern is provided via command line option `-p`

When I ran my script, less reported the following message
> "No previous regular expression  (press RETURN)"

I guess it (less) executed the `+jgn` actions too early. it should have processed the `-p` option first, rather than complaining a missing or empty lesshst file

Then I pressed the RETURN key, got the same message, then RETURN again, it finally did the search

It seems when the message **No previous regular expression  (press RETURN)** appears the first time, and `/` is pressed, it would trigger the bug as well.
